samples/unit-tests/series/datagrouping/demo.js

```diff
@@ -825,6 +825,76 @@ QUnit.test('When groupAll: true, group point should have the same start regardle
         chart.series[1].points[1].dataGroup.start,
         'When the groupAll: false, and new extremes don\t influence the group, the start should not be changed.'
     );
+
+    // Change the data set to check the group start,
+    // when the extremes overlap with the point x.
+    chart.series[0].update({
+        type: 'line',
+        dataGrouping: {
+            groupAll: true,
+            units: [
+                ['minute', [5]]
+            ]
+        },
+        data: [
+            [1610028057000, 0.25],
+            [1610033040000, 0.80125],
+            [1610118031000, 0.8475],
+            [1610118209000, 0.8475],
+            [1610118426000, 0.8475],
+            [1610118691000, 0.8475],
+            [1610120241000, 0.8475],
+            [1610372248000, 0.8325],
+            [1610373264000, 0.83],
+            [1610373445000, 0.8275],
+            [1610384401000, 0.835],
+            [1610384401000, 0.835],
+            [1610392040000, 0.375],
+            [1610719978000, 0.915],
+            [1610720025000, 0.915],
+            [1610724043000, 0.91],
+            [1610724275000, 0],
+            [1610725033000, 0.9],
+            [1610725069000, 0.9],
+            [1610729723000, 0],
+            [1611071398000, 0.84375],
+            [1611138383000, 0.835],
+            [1611159135000, 0.77],
+            [1611162097000, 0.7825],
+            [1611162097000, 0.7825]
+        ]
+    }, false);
+    chart.series[1].remove();
+
+    const point = chart.series[0].points[10],
+        pointX = point.x;
```

Comment:
You defined 2 variables that you are using only once, and below, you are using `chart.series[0].points[9]` few times. You could either use variables more efficiently, or (which I prefer in this case) you could not use variables at all.